### ISSUE TITLE
BackgroundExecutingPresenter

### ISSUE THREAD

**Issue by rhmndnt** [MrHadiSatrio/Journal3#2]
https://github.com/MrHadiSatrio/Journal3/blob/30ff779807a5e171d9b86fb0d8e463836fa27039/app-android-journal3/src/main/kotlin/com/hadisatrio/apps/android/journal3/story/ViewStoriesActivity.kt#L52

Could we do the same as `BackgroundExecutingUseCase` for the Presenter?

**Comment by MrHadiSatrio** [comment]
I'm having a hard time to visualize the use-case for this. Could you give an example where such decorator might be useful?

**Comment by rhmndnt** [comment]
```kotlin
 presenter = BackgroundExecutingPresenter(
       scope = lifecycleScope,
       origin = CachingStoriesPresenter(
          origin = MainThreadEnforcingPresenter(
              StoriesRecyclerViewPresenter(
                  recyclerView = findViewById(R.id.stories_list)
              )
          )
      )
  ),
```

Instead
```kotlin
   presenter = CachingStoriesPresenter(
        scope = lifecycleScope + Dispatchers.Default,
        origin = MainThreadEnforcingPresenter(
            StoriesRecyclerViewPresenter(
                recyclerView = findViewById(R.id.stories_list)
            )
        )
    ),
```

**Comment by MrHadiSatrio** [comment]
Ah, I see. Indeed that will bring a nice symmetry between `Stories` and `Presenter`! Thanks for your suggestion. I'll take this up as API enhancements together with #1.

(Do feel free if you want to contribute this as well, just let me know. :) )

**Comment by MrHadiSatrio** [comment]
This has been resolved with 4a0389c82510339a12ed8b64f4cb23268063b4f5; we now have `CoroutineDispatchingPresenter` which works similarly to your suggestion when instantiated with `Dispatchers.Default`. Similarly, it also replaces `MainThreadEnforcingPresenter` when instantiated with `Dispatchers.Main`.

Thanks for reporting!
